File: bindings/lua/Polycode/AABB.lua
class "AABB"


function AABB:__getvar(name)
	if name == "min" then
		local retVal = Polycode.AABB_get_min(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "max" then
		local retVal = Polycode.AABB_get_max(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function AABB:__setvar(name,value)
	if name == "min" then
		Polycode.AABB_set_min(self.__ptr, value.__ptr)
		return true
	elseif name == "max" then
		Polycode.AABB_set_max(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function AABB:__delete()
	if self then Polycode.delete_AABB(self.__ptr) end
end

File: bindings/javascript/Polycode/AABB.js
function AABB() {
	Object.defineProperties(this, {
		'min': { enumerable: true, configurable: true, get: AABB.prototype.__get_min, set: AABB.prototype.__set_min},
		'max': { enumerable: true, configurable: true, get: AABB.prototype.__get_max, set: AABB.prototype.__set_max}
	})
}
AABB.prototype.__get_min = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.AABB__get_min(this.__ptr)
	return retVal
}

AABB.prototype.__set_min = function(val) {
	Polycode.AABB__set_min(this.__ptr, val.__ptr)
}

AABB.prototype.__get_max = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.AABB__get_max(this.__ptr)
	return retVal
}

AABB.prototype.__set_max = function(val) {
	Polycode.AABB__set_max(this.__ptr, val.__ptr)
}

Duktape.fin(AABB.prototype, function (x) {
	if (x === AABB.prototype) {
		return;
	}
	Polycode.AABB__delete(x.__ptr)
})
